### ISSUE TITLE
XGBoost early stopping - group aware validation set

### ISSUE THREAD

**Issue by klin333** [tidymodels/parsnip#471]
When not all samples are independent, eg one customer having multiple samples, I wish to use group aware partitioning of train/validation sets, ie samples from one customer to only appear in train or validation but not both. For cross validation, this can be done by rsample::group_vfold_cv()

However, for xgboost early stopping purposes, it appears the current implementation does a simple random sample out of the samples given to parsnip::xgb_train(). Help on ideas to enable group aware validation set partitioning for early stopping?

**Comment by simonpcouch** [comment]
Thanks for the issue, and apologies for the delay. Closing in favor of duplicate #765—future discussion will reside there. :)
